config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-presubmits-main.yaml

```diff
@@ -80,83 +80,57 @@ presets:
     spec:
       containers:
-      - image: gcr.io/cluster-api-provider-vsphere/extra/mdlint:0.17.0
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230719-72ce785784-1.27
         command:
-        - /nodejs/bin/node
+        - runner.sh
         args:
-        - /md/lint
-        - -i
-        - vendor
-        - -i
-        - contrib/haproxy/openapi
-        - .
-    annotations:
-      testgrid-dashboards: vmware-cluster-api-provider-vsphere, sig-cluster-lifecycle-cluster-api-provider-vsphere
-      testgrid-tab-name: pr-verify-markdown
-      testgrid-alert-email: sig-cluster-lifecycle-cluster-api-vsphere-alerts@kubernetes.io
-      description: Verifies the markdown has been linted
-
-  - name: pull-cluster-api-provider-vsphere-verify-shell
-    always_run: false
-    run_if_changed: '.*\.\w*sh$'
-    decorate: true
-    path_alias: sigs.k8s.io/cluster-api-provider-vsphere
-    spec:
-      containers:
-      - image: gcr.io/cluster-api-provider-vsphere/extra/shellcheck:v0.6.0
-        command:
-        - /bin/shellcheck.sh
+        - ./hack/ci-apidiff.sh
     annotations:
       testgrid-dashboards: vmware-cluster-api-provider-vsphere, sig-cluster-lifecycle-cluster-api-provider-vsphere
-      testgrid-tab-name: pr-verify-shell
+      testgrid-tab-name: pr-apidiff-main
       testgrid-alert-email: sig-cluster-lifecycle-cluster-api-vsphere-alerts@kubernetes.io
-      description: Verifies the shell scripts have been linted
+      description: Checks for API changes in the PR
 
   - name: pull-cluster-api-provider-vsphere-verify-main
+    branches:
+    - ^main$
+    labels:
+      preset-dind-enabled: "true"
     always_run: true
     decorate: true
     path_alias: sigs.k8s.io/cluster-api-provider-vsphere
     spec:
       containers:
       - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230719-72ce785784-1.27
         command:
+        - runner.sh
+        args:
         - make
         - verify
+        # we need privileged mode in order to do docker in docker
+        securityContext:
+          privileged: true
     annotations:
       testgrid-dashboards: vmware-cluster-api-provider-vsphere, sig-cluster-lifecycle-cluster-api-provider-vsphere
-      testgrid-tab-name: pr-verify
+      testgrid-tab-name: pr-verify-main
       testgrid-alert-email: sig-cluster-lifecycle-cluster-api-vsphere-alerts@kubernetes.io
 
-  - name: pull-cluster-api-provider-vsphere-test
+  - name: pull-cluster-api-provider-vsphere-test-main
     branches:
-    # The script this job runs is not in all branches.
     - ^main$
     always_run: false
-    run_if_changed: '^((api|cmd|config|contrib|controllers|pkg|test)/|Makefile|go\.mod|go\.sum|main\.go)'
+    # Run if go files, scripts or configuration changed (we use the same for all jobs for simplicity).
+    run_if_changed: '^((apis|config|controllers|feature|hack|packaging|pkg|test|webhooks)/|Dockerfile|go\.mod|go\.sum|main\.go|Makefile)'
```

Comment:
Please note that the webhooks pkg doesn't exist yet. But it is very likely to be introduced given ongoing work in core CAPI to reduce dependencies of api packges that we will probalby also align to in CAPV: https://github.com/kubernetes-sigs/cluster-api/issues/9011